Sources/Xcodeproj/pbxproj().swift

```diff
@@ -164,6 +164,7 @@ func xcodeProject(
     projectSettings.debug.GCC_OPTIMIZATION_LEVEL = "0"
     projectSettings.debug.ONLY_ACTIVE_ARCH = "YES"
     projectSettings.debug.SWIFT_OPTIMIZATION_LEVEL = "-Onone"
+    projectSettings.debug.SWIFT_ACTIVE_COMPILATION_CONDITIONS += ["DEBUG"]
```

Comment:
We should add this define in SwiftPM's build system too. This can go somewhere around here https://github.com/apple/swift-package-manager/blob/master/Sources/Build/BuildPlan.swift#L306

Reply:
This should be:
```swift
projectSettings.debug.SWIFT_ACTIVE_COMPILATION_CONDITIONS = ["$(inherited)", "DEBUG"]
```

Reply:
I guess this applies to the `OTHER_SWIFT_FLAGS` too?

Reply:
Looks like `OTHER_SWIFT_FLAGS` already has that entry https://github.com/apple/swift-package-manager/blob/master/Sources/Xcodeproj/pbxproj().swift#L454

Reply:
I think we should move `-DXcode` from OTHER_SWIFT_FLAGS to this build setting but that is orthogonal to this PR.